src/GCDHelper.cs

```diff
@@ -250,14 +264,71 @@ private void ResetBar(Configuration conf) {
             SlideStart_RightTri = false;
             SlideEnd_RightTri = false;
             Slide_Background = false;
+            triggeredAlerts.Clear();
+        }
+
+        private bool CheckAlert(EventType type, EventCause cause) {
+            string key = $"{type}-{cause}";
+            return triggeredAlerts.ContainsKey(key) && triggeredAlerts[key];
+        }
+
+        private void MarkAlert(EventType type, EventCause cause) {
+            string key = $"{type}-{cause}";
+            triggeredAlerts[key] = true;
+        }
+
+        private void PulseCheckerSlide(BarInfo bar, Configuration conf){
```

Comment:
Let's avoid tying the events to alert types. Something like `CheckSlideEvent` and `CheckQueueEvent`

Reply:
Good idea.

Reply:
Done.